packages/webapp/src/pages/chat/Chat.tsx

```diff
@@ -59,13 +65,127 @@ const Chat = () => {
     setExcludeCategory(newValue || '');
   };
 
+  const onEnableBrandingChange = (_event?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: boolean) => {
+    setEnableBranding(!!checked);
+  };
+
   const onUseSuggestFollowupQuestionsChange = (
     _event?: React.FormEvent<HTMLElement | HTMLInputElement>,
     checked?: boolean,
   ) => {
     setUseSuggestFollowupQuestions(!!checked);
   };
 
+  const [isDarkTheme, setIsDarkTheme] = useState(() => {
+    const storedTheme = localStorage.getItem('isDarkTheme');
+    return storedTheme ? JSON.parse(storedTheme) : false;
+  });
+
+  const [customStyles, setCustomStyles] = useState(() => {
+    const styleDefaultsLight = {
+      AccentHigh: '#692b61',
+      AccentLight: '#f6d5f2',
+      AccentDark: '#5e3c7d',
+      TextColor: '#123f58',
+      BackgroundColor: '#e3e3e3',
+      ForegroundColor: '#4e5288',
+      FormBackgroundColor: '#f5f5f5',
+      BorderRadius: '10px',
+      BorderWidth: '3px',
+      FontBaseSize: '14px',
+    };
+
+    const styleDefaultsDark = {
+      AccentHigh: '#dcdef8',
+      AccentLight: '#032219',
+      AccentDark: '#fdfeff',
+      TextColor: '#fdfeff',
+      BackgroundColor: '#e3e3e3',
+      ForegroundColor: '#4e5288',
+      FormBackgroundColor: '#32343e',
+      BorderRadius: '10px',
+      BorderWidth: '3px',
+      FontBaseSize: '14px',
+    };
+    const defaultStyles = isDarkTheme ? styleDefaultsDark : styleDefaultsLight;
+    const storedStyles = localStorage.getItem('customStyles');
+    return storedStyles ? JSON.parse(storedStyles) : defaultStyles;
+  });
+
+  const handleCustomStylesChange = (newStyles: CustomStylesState) => {
+    setCustomStyles(newStyles);
+  };
+
+  const handleThemeToggle = (newIsDarkTheme: boolean) => {
+    // Get the ChatComponent instance (modify this according to how you manage your components)
+    const chatComponent = document.querySelector('chat-component');
+    if (chatComponent) {
+      // Remove existing style attributes
+      chatComponent.removeAttribute('style');
+      // eslint-disable-next-line unicorn/prefer-dom-node-dataset
+      chatComponent.setAttribute('data-theme', newIsDarkTheme ? 'dark' : '');
+    }
+    // Update the body class and html data-theme
+    localStorage.removeItem('customStyles');
+    document.body.classList.toggle('dark', newIsDarkTheme);
+    document.documentElement.dataset.theme = newIsDarkTheme ? 'dark' : '';
```

Comment:
this should probably use reactive update of the React component rather than direct class add